server/schedule/checker/replica_strategy.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	"github.com/pingcap/log"
+	"github.com/pingcap/pd/v4/server/core"
+	"github.com/pingcap/pd/v4/server/schedule/filter"
+	"github.com/pingcap/pd/v4/server/schedule/opt"
+	"github.com/pingcap/pd/v4/server/schedule/selector"
+	"go.uber.org/zap"
+)
+
+// ReplicaStrategy collects some utilities to manipulate region peers. It
+// exists to allow replica_checker and rule_checker to reuse common logics.
+type ReplicaStrategy struct {
+	checkerName    string // replica-checker / rule-checker
+	cluster        opt.Cluster
+	locationLabels []string
+	region         *core.RegionInfo
+	extraFilters   []filter.Filter
+}
+
+// SelectStoreToAdd returns the store to add a replica to a region.
+// `coLocationStores` are the stores used to compare location with target
+// store.
+// `extraFilters` is used to set up more filters based on the context that
+// calling this method.
+//
+// For example, to select a target store to replace a region's peer, we can use
+// the peer list without the peer to be removed as `coLocationStores`.
```

Comment:
> the peer list without the peer to be removed as `coLocationStores`.

What's the meaning of this sentence?

Reply:
For example, region contains 3 peers `[p1, p2, p3]`. In order to replace `p3`, we will call `SelectStoreToAdd` and pass `[p1, p2]` in.

Reply:
I see...So this sentence like this maybe better?

> the peer list with the peer removed as `coLocationStores`.

Reply:
good idea. thanks.